3rdparty/glsl-optimizer/src/glsl/ir_print_glsl_visitor.cpp

```diff
@@ -50,7 +50,7 @@ static inline const char* get_precision_string (glsl_precision p)
 static const int tex_sampler_type_count = 7;
 // [glsl_sampler_dim]
 static const char* tex_sampler_dim_name[tex_sampler_type_count] = {
-	"1D", "2D", "3D", "Cube", "Rect", "Buf", "External",
+	"1D", "2D", "3D", "Cube", "Rect", "Buf", "2D",
```

Comment:
"2D" already appears in array...

Reply:
The glsl-optimizer replaced all texture lookups in the shader that used a "uniform samplerExternalOES" with the function "textureExternal" (which is deprecated) instead of "texture2D" (which also supportes samplerExternal).

e.g. the input:
```
uniform lowp samplerExternalOES s_texColor;
...
gl_FragColor = texture2D(s_texColor, v_texcoord0)
```
would become: 
```
uniform lowp samplerExternalOES s_texColor;
...
gl_FragColor = textureExternal(s_texColor, v_texcoord0)
```
I felt it was easiest to just change the name of the sampler in the array since the rest was handled correctly